src/Microsoft.ML.Transforms/NAIndicatorTransform.cs

```diff
@@ -68,322 +76,549 @@ private static VersionInfo GetVersionInfo()
-                type, LoaderSignature);
-        }
-
-        private const string RegistrationName = "NaIndicator";
-
-        // The output column types, parallel to Infos.
-        private readonly ColumnType[] _types;
+        private const string RegistrationName = nameof(NAIndicatorTransform);
 
         /// <summary>
-        /// Convenience constructor for public facing API.
+        /// Initializes a new instance of <see cref="NAIndicatorTransform"/>
         /// </summary>
-        /// <param name="env">Host Environment.</param>
-        /// <param name="input">Input <see cref="IDataView"/>. This is the output from previous transform or loader.</param>
-        /// <param name="name">Name of the output column.</param>
-        /// <param name="source">Name of the column to be transformed. If this is null '<paramref name="name"/>' will be used.</param>
-        public NAIndicatorTransform(IHostEnvironment env, IDataView input, string name, string source = null)
-            : this(env, new Arguments() { Column = new[] { new Column() { Source = source ?? name, Name = name } } }, input)
+        /// <param name="env">The environment to use.</param>
+        /// <param name="columns">The names of the input columns of the transformation and the corresponding names for the output columns.</param>
+        internal NAIndicatorTransform(IHostEnvironment env, params (string input, string output)[] columns)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(NAIndicatorTransform)), columns)
         {
         }
 
-        /// <summary>
-        /// Public constructor corresponding to SignatureDataTransform.
-        /// </summary>
-        public NAIndicatorTransform(IHostEnvironment env, Arguments args, IDataView input)
-            : base(env, RegistrationName, Contracts.CheckRef(args, nameof(args)).Column,
-                input, TestType)
+        internal NAIndicatorTransform(IHostEnvironment env, Arguments args)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(NAIndicatorTransform)), GetColumnPairs(args.Column))
         {
-            Host.AssertNonEmpty(Infos);
-            Host.Assert(Infos.Length == Utils.Size(args.Column));
-
-            _types = GetTypesAndMetadata();
         }
 
-        private NAIndicatorTransform(IHost host, ModelLoadContext ctx, IDataView input)
-            : base(host, ctx, input, TestType)
+        private NAIndicatorTransform(IHostEnvironment env, ModelLoadContext ctx)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(NAIndicatorTransform)), ctx)
         {
-            Host.AssertValue(ctx);
-            Host.AssertNonEmpty(Infos);
+            Host.CheckValue(ctx, nameof(ctx));
+        }
 
-            // *** Binary format ***
-            // <base>
+        private static (string input, string output)[] GetColumnPairs(Column[] columns)
```

Comment:
>GetColumnPairs [](start = 55, length = 14)

this entire method can be replaced with
`columns.Select(c=>(c.Source ?? c.Name, c.Name)).ToArray()`